### PR TITLE
Basic Go Programs - Day 7

### PR DESCRIPTION
**Pull request by mrniranjan**
Loops

Signed-off-by: Niranjan Mallapadi <niranjan@ashoo.in>

### CODE REVIEW

File: loops/loop26.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+package main
+
+import (
+	"fmt"
+)
+
+func main() {
+	number := 980
+ 	var factors, primeFactors = []int{}, []int{}
+	for i :=1; i < number; i++ {
```
Comment (gsr-shanks) [inline loops/loop26.go]: i <= number

File: loops/loop25.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	}
+	return digits
+}
+
+func sumOfDigits(arr []int) int {
+	sum := 0
+	for _, v := range arr {
+		sum += v
+	}
+	return sum
```
Comment (gsr-shanks) [inline loops/loop25.go]: let's have them as 'Named return values'

File: loops/loop25.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+	}
+}
+
+func getDigits(n int) []int {
+	digits := make([]int, 0, 1)
+	for i :=0 ; n != 0; i++ {
+		digits = append(digits, n % 10)
+		n = n / 10
+	}
+	return digits
```
Comment (gsr-shanks) [inline loops/loop25.go]: let's have them as 'Named return values'

File: loops/loop27.go [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+package main
+
+import "fmt"
+
+func main() {
+	arr1 := []int{3,4,5,6}
+	arr2 := []int{3,6,7,8}
+	for i:=1; i < len(arr1); i++ {
+		for j :=1; j < len(arr2); j++ {
+			if (i * j  >= 0) {
```
Comment (gsr-shanks) [inline loops/loop27.go]: You don't need brackets for this 'if' statement.

File: loops/loop30.go [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+package main
+
+import "fmt"
+
+func main() {
+	numbers := []int{2,3,5,9,18,20,25,27,29,35,46,}
+	for i := range numbers {
+		if numbers[i] % 2 == 0 {
+			if numbers[i] % 3 == 0 {
+				fmt.Printf("%d is divisible by 2 and 6\n", numbers[i])
```
Comment (gsr-shanks) [inline loops/loop30.go:12]: You mean '2 and 3' here I guess.

File: loops/loop30.go [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+			}
+		}
+	}
+	// another method
+
+	for i := range numbers {
+		if numbers[i] %2 == 0 && numbers[i] % 3 == 0 {
+			fmt.Printf("%d is divisible by 2 and 6\n", numbers[i])
```
Comment (gsr-shanks) [inline loops/loop30.go:20]: You mean '2 and 3' here I guess.

File: loops/loop30.go (hunk 5 above)
Comment (mrniranjan) [reply loops/loop30.go:12]: No it's 2 and 6 (well 3 also. but in this if a number is divisible by 2 and 3, it's divisible by 6.)

**Review by mrniranjan — COMMENTED** [review]
Made requested changes

File: loops/loop30.go (hunk 5 above)
Comment (gsr-shanks) [reply loops/loop30.go:12]: Ok, so that would be 2, 3 and 6 then. :)

**Comment by gsr-shanks** [comment]
LGTM

**Pull request merged**